src/hooks/useArNSDomainPriceList.tsx

```diff
@@ -0,0 +1,53 @@
+import { useGlobalState } from '@src/state';
+import { isARNSDomainNameValid, lowerCaseDomain } from '@src/utils';
+import eventEmitter from '@src/utils/events';
+import { useQuery } from '@tanstack/react-query';
+
+export function useArNSDomainPriceList(domain: string) {
+  const [{ arioContract, ioProcessId }] = useGlobalState();
+
+  return useQuery({
+    queryKey: [`arnsDomainPriceList-${domain.length}`, ioProcessId.toString()],
+    queryFn: async () => {
+      const prices: {
+        lease: number; // lease of 1 year
+        buy: number; // permabuy
+      } = {
+        lease: 0,
+        buy: 0,
+      };
+      try {
+        if (
+          domain &&
+          domain.length > 0 &&
+          isARNSDomainNameValid({ name: domain })
+        ) {
+          const sharedOptions: any = {
+            intent: 'Buy-Record',
+            name: lowerCaseDomain(domain),
+          };
+          const [leasePrice, buyPrice] = await Promise.all([
+            arioContract.getTokenCost({
+              ...sharedOptions,
+              years: 1,
+              purchaseType: 'lease',
+            }),
+            arioContract.getTokenCost({
+              ...sharedOptions,
+              purchaseType: 'permabuy',
+            }),
+          ]);
+          prices.lease = leasePrice;
+          prices.buy = buyPrice;
+        }
+      } catch (error) {
+        eventEmitter.emit('error', {
+          message: `Failed to fetch price for domain: ${domain}`,
+          name: 'Domain Price',
+        });
+      }
+      return prices;
+    },
+    staleTime: 60 * 1000, // 1 minute
```

Comment:
you can trust prices for up to a full demand factor period (1-day) - so i'd say it's okay to cache this to something more like 5-15 mins

Reply:
ahhh good to know

Reply:
Will note to include when the next DF update is in the pricelist ticket